airflow/settings.py

```diff
@@ -61,28 +32,38 @@ def gauge(cls, stat, value, rate=1, delta=False):
  """
 
 BASE_LOG_URL = '/admin/airflow/log'
-AIRFLOW_HOME = os.path.expanduser(conf.get('core', 'AIRFLOW_HOME'))
-SQL_ALCHEMY_CONN = conf.get('core', 'SQL_ALCHEMY_CONN')
 LOGGING_LEVEL = logging.INFO
-DAGS_FOLDER = os.path.expanduser(conf.get('core', 'DAGS_FOLDER'))
-
-engine_args = {}
-if 'sqlite' not in SQL_ALCHEMY_CONN:
-    # Engine args not supported by sqlite
-    engine_args['pool_size'] = conf.getint('core', 'SQL_ALCHEMY_POOL_SIZE')
-    engine_args['pool_recycle'] = conf.getint('core',
-                                              'SQL_ALCHEMY_POOL_RECYCLE')
-
-engine = create_engine(SQL_ALCHEMY_CONN, **engine_args)
-Session = scoped_session(
-    sessionmaker(autocommit=False, autoflush=False, bind=engine))
 
 # can't move this to conf due to ConfigParser interpolation
 LOG_FORMAT = (
     '[%(asctime)s] {%(filename)s:%(lineno)d} %(levelname)s - %(message)s')
 SIMPLE_LOG_FORMAT = '%(asctime)s %(levelname)s - %(message)s'
 
 
+engine = None
+Session = None
+
+
+def connect(sql_alchemy_conn, pool_size, pool_recycle):
+    """
+    builds the Session object to connect to the SQL-alchemy backend.
+    """
+    global Session, engine
+
+    if Session:
+        Session.remove()
+
+    engine_args = {}
+    if 'sqlite' not in sql_alchemy_conn:
+        # Engine args not supported by sqlite
+        engine_args['pool_size'] = pool_size
+        engine_args['pool_recycle'] = pool_recycle
+
+    engine = create_engine(sql_alchemy_conn, **engine_args)
+    Session = scoped_session(sessionmaker(autocommit=False, autoflush=False,
+                                          bind=engine))
+
```

Comment:
Move the creation of the sql alchemy connection inside this method, which get called whenever a new configuration is loaded (in the `load_config()`method of `airflow/configuration.py`)